app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/SimpleEndpointStepHandler.java

```diff
@@ -0,0 +1,112 @@
+import io.syndesis.model.integration.SimpleStep;
+import io.syndesis.model.integration.Step;
+import org.apache.camel.model.ProcessorDefinition;
+import org.apache.camel.runtimecatalog.RuntimeCamelCatalog;
+import org.apache.camel.util.ObjectHelper;
+
+/**
+ * This is needed until connectors are migrated to the new architecture.
+ */
+public class SimpleEndpointStepHandler implements IntegrationStepHandler {
+    @Override
+    public boolean canHandle(Step step) {
+        if (!"endpoint".equals(step.getStepKind()) && !"connector".equals(step.getStepKind())) {
+            return false;
+        }
+
+        if (step.getConnection().isPresent()) {
+            return false;
+        }
+        if (!step.getAction().filter(ConnectorAction.class::isInstance).isPresent()) {
+            return false;
+        }
+
+        ConnectorAction action = step.getAction().filter(ConnectorAction.class::isInstance).map(ConnectorAction.class::cast).get();
+
+        if (action.getDescriptor() == null) {
+            return false;
+        }
+
+        return action.getDescriptor().getComponentScheme().isPresent();
+    }
+
+    @SuppressWarnings({"unchecked", "PMD"})
+    @Override
+    public Optional<ProcessorDefinition> handle(Step step, ProcessorDefinition route, IntegrationRouteBuilder builder) {
+        // Model
+        final ConnectorAction action = step.getAction().filter(ConnectorAction.class::isInstance).map(ConnectorAction.class::cast).get();
+        final ConnectorDescriptor descriptor = action.getDescriptor();
+
+        // Camel
+        final String index = step.getMetadata(Step.METADATA_STEP_INDEX).orElseThrow(() -> new IllegalArgumentException("Missing index for step:" + step));
+        final String componentScheme = action.getDescriptor().getComponentScheme().get();
+        final Map<String, String> configuredProperties = step.getConfiguredProperties();
+        final Map<String, String> properties = action.filterEndpointProperties(configuredProperties);
+
+        properties.entrySet()
+            .stream()
+            .filter(action::isEndpointProperty)
+            .filter(action::isSecret)
+            .forEach(e -> e.setValue(String.format("{{%s-%s.%s}}", componentScheme, index, e.getKey())));
+
+        // any configuredProperties on action descriptor are considered
+        properties.putAll(descriptor.getConfiguredProperties());
+
+        try {
+            final RuntimeCamelCatalog catalog = builder.getContext().getRuntimeCamelCatalog();
+            final String uri = catalog.asEndpointUri(componentScheme, Map.class.cast(properties), false);
+
+            if (route == null) {
+                route = builder.from(uri);
+            } else {
+                route = route.to(uri);
+            }
+        } catch (URISyntaxException e) {
+            throw ObjectHelper.wrapRuntimeCamelException(e);
+        }
+
+        // Handle split
+        if (descriptor.getSplit().isPresent()) {
```

Comment:
see ConnectorStepHandler for the same code.

Reply:
done